File: Data Structures & Algorithms/subsets/submission-3.java
class Solution {
    List<List<Integer>> res = new ArrayList<>();
    public List<List<Integer>> subsets(int[] nums) {
        res = new ArrayList<>();
        dfs(new ArrayList<>(), nums, 0);
        return res;
    }

    public void dfs(List<Integer> curr, int[] nums, int index){
        if(index >= nums.length){
            res.add(new ArrayList<>(curr));
            return;
        }

        curr.add(nums[index]);
        dfs(curr, nums, index + 1);

        curr.remove(curr.size() - 1);
        dfs(curr, nums, index + 1);
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque


class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:

        dq = deque()
        res = []

        if not root:
            return res

        dq.append(root)

        while dq:
            level_size = len(dq)

            for index in range(level_size):
                curr = dq.popleft()

                if curr.left:
                    dq.append(curr.left)
                if curr.right:
                    dq.append(curr.right)

                if index == level_size - 1:
                    res.append(curr.val)
        return res
